From 2dae7ee6765d8fad100172fba7021b2f51706c7a Mon Sep 17 00:00:00 2001
From: agaraix <anath.bandux.garai@intel.com>
Date: Thu, 9 Jan 2020 18:48:34 +0530
Subject: [PATCH] UGW_SW-45728 : PPA klocwork fix

Explicit check to avoid out-of-bound array index access.
---
 drivers/net/ethernet/lantiq/ppa/ppa_hal_wrapper.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/ppa/ppa_hal_wrapper.c b/drivers/net/ethernet/lantiq/ppa/ppa_hal_wrapper.c
index 18d3b2cd7398..f1f1d6d6381f 100644
--- a/drivers/net/ethernet/lantiq/ppa/ppa_hal_wrapper.c
+++ b/drivers/net/ethernet/lantiq/ppa/ppa_hal_wrapper.c
@@ -152,6 +152,9 @@ uint32_t ppa_drv_register_cap(PPA_API_CAPS cap, uint8_t wt, PPA_HAL_ID hal_id)
 {
 	PPA_HSEL_CAP_NODE *cap_node;
 
+	if (cap >= MAX_CAPS)
+		return PPA_FAILURE;
+
 	/* allocate the node*/
 	cap_node  =  (PPA_HSEL_CAP_NODE*) kmalloc (sizeof(PPA_HSEL_CAP_NODE), GFP_KERNEL);
 	if (!cap_node) {
@@ -174,6 +177,9 @@ uint32_t ppa_drv_deregister_cap(PPA_API_CAPS cap, PPA_HAL_ID hal_id)
 	PPA_HSEL_CAP_NODE *t_node = NULL;
 	PPA_HLIST_NODE *tmp;
 
+	if (cap >= MAX_CAPS)
+		return PPA_FAILURE;
+
 	ppa_hlist_for_each_entry_safe(t_node, tmp, &g_hsel_caps_list[cap], cap_list) {
 		if (t_node->hal_id  ==  hal_id) {
 			ppa_hlist_del(&t_node->cap_list);
